API 7 bump

reiichi001 opened the pull request:
- Removes duplicate package references and groups the NuGet ones together.
- bumped to Net 6
- added repo url

reiichi001 committed:
API 7 bump

- Removes duplicate package references and groups the NuGet ones together.
- bumped to Net 6
- added repo url

jacksonrakena commented:
@reiichi001 would it be worth updating all the dependencies in this pull request? For some reason the RPC plugin randomly doesn't work for a subset of users and updating the dependencies (I *believe* it's XivCommon that is the culprit) fixes that issue.

Updating the dependencies fixed [this issue with null data](https://github.com/goaaats/Dalamud.RichPresence/issues/17#issuecomment-1272960293).

I fixed all those issues in my fork [Abyssal.RichPresence](https://github.com/jacksonrakena/Abyssal.RichPresence) but my fork has diverged quite a bit (I did some additional fixes with DiscordRPC, [which apparently fails to connect sometimes because of a JSON parsing issue](https://github.com/Lachee/discord-rpc-csharp/issues/214)) so it might be worth just updating this PR.

Updating dependencies would fix #17 and #12, I believe.

reiichi001 commented:
@jacksonrakena, we're already using the latest stable version of the discord-rpc-csharp package. If you want to pull their current master, that should be done via submoduling, not pulling in the source, building it in debug, and then referencing the DLL. See <https://github.com/reiichi001/Dalamud.RichPresence/tree/submodule-discordrpc> as an example of how to do this.

While I can definitely update the other dependencies like XivCommon and push a new build for that, I'm hesitant to switch to submoduling the RPC library as we're then basically pulling a nonrelease from a random snapshot in time that could have bugs or other WIP code. I'd personally want to wait for an updated NuGet package, or at a minimum, a tagged release which could then be submoduled.

reiichi001 committed:
Update dependencies

jacksonrakena commented:
> @jacksonrakena, we're already using the latest stable version of the discord-rpc-csharp package. If you want to pull their current master, that should be done via submoduling, not pulling in the source, building it in debug, and then referencing the DLL. See https://github.com/reiichi001/Dalamud.RichPresence/tree/submodule-discordrpc as an example of how to do this.
> 
> While I can definitely update the other dependencies like XivCommon and push a new build for that, I'm hesitant to switch to submoduling the RPC library as we're then basically pulling a nonrelease from a random snapshot in time that could have bugs or other WIP code. I'd personally want to wait for an updated NuGet package, or at a minimum, a tagged release which could then be submoduled.

@reiichi001 Ah yes, I should’ve made it clear that I wasn’t intending to PR from my fork, the pulling was just a quick hack to see if that was the issue.

I filed [an issue](https://github.com/Lachee/discord-rpc-csharp/issues/214) in DiscordRPC so hopefully it is fixed and another NuGet release made.

jacksonrakena commented:
@reiichi001 DiscordRPC just patched the serialization error and published a release (1.1.1.14), if you want to consider updating? [https://www.nuget.org/packages/DiscordRichPresence/](https://www.nuget.org/packages/DiscordRichPresence/)

reiichi001 committed:
Update discord-rpc library

Bigger version number jump for the more important dependency.

reiichi001 committed:
Adjust deps

reiichi001 committed:
Updates for NET7 / API 8

keysmashes committed:
Allow hiding rich presence entirely when AFK

NPittinger committed:
refactor: update for Dalamud v9

reiichi001 committed:
Merge pull request #2 from NPittinger/dev

Update for Dalamud API 9

reiichi001 committed:
Merge pull request #1 from sersorrel/afk-hides-status

Allow hiding rich presence entirely when AFK

WorkingRobot committed:
New IPC; Update to .NET 8

KazWolfe committed:
Bump to API 10 / DT support

reiichi001 committed:
Merge pull request #4 from KazWolfe/dt

reiichi001 committed:
Prepare for release

Merged in Kaz's PR.
Bump version.
Fixed build issues.

reiichi001 committed:
Merge pull request #3 from WorkingRobot/master

reiichi001 committed:
Bump for IPC update

Also moves the config window close function to after config is saved.

reiichi001 committed:
Switch to ScaledVector2

Blooym committed:
feat: Wine support + use Dalamud.NET.Sdk

reiichi001 committed:
Merge pull request #5 from Blooym/add-wine-support

reiichi001 committed:
Add Linux Wine IPC Bridge baked in

Thanks @Blooym

reiichi001 committed:
Bake the exe in because build system can't pull it from online

Pushing so I can ask someone on Linux to test and verify it's working as intended before I push a PR update.

Also maybe fix the build warnings on local because Visual Studio was getting on my nerves.

reiichi001 committed:
Update to API 11

Everything should work now that LocalPlayer is good. I might need someone with non-English to verify AFK mode.

reiichi001 committed:
Fix image display bug

Credit to @Shinryin 's PR that handled something I missed. I cherry picked a few things and more looks to be working.

reiichi001 committed:
Fix a typo that broke full job names

reiichi001 committed:
Maybe fix parties

reiichi001 committed:
Only spit out debug statements for debug builds

reiichi001 committed:
bump version

Blooym committed:
Bump wine-discord-ipc-bridge to 0.0.3

reiichi001 committed:
Merge pull request #7 from Blooym/master

reiichi001 committed:
Disable assets until discord-rpc-csharp fixes them

https://github.com/Lachee/discord-rpc-csharp/issues/267

reiichi001 committed:
Disable the rest of the assets

reiichi001 committed:
Enable assets

Discord fixed their issue.

reiichi001 committed:
update WineRPCBridge binary

This is wine-discord-ipc-bridge compiled locally without the compiler optimization that makes windows AVs angry.

https://www.virustotal.com/gui/file/14668f1c815a421a7cc77da4d39659142b56292600db6cb004ce16218193ff0c

reiichi001 committed:
API 12

reiichi001 committed:
API 13

reiichi001 committed:
API 14

vmfunc committed:
feat: add option to always show home world in details

vmfunc committed:
feat: show data center name alongside world

vmfunc committed:
feat: hide presence during cutscenes

reiichi001 committed:
Merge pull request #9 from vmfunc/master

reiichi001 committed:
Merge pull request #10 from vmfunc/feat/show-datacenter

reiichi001 committed:
Merge pull request #11 from vmfunc/feat/hide-in-cutscene
